xbmc/application/Application.cpp

```diff
@@ -2430,29 +2412,76 @@ bool CApplication::PlayFile(CFileItem item,
     }
   }
 
-  // a disc image might be Blu-Ray disc
+  // Now item needs to point to the item we are playing (ie. the part of the stack)
+  if (isStack)
+  {
+    // Save info needed from stack item
+    bool Resume{item.GetStartOffset() == STARTOFFSET_RESUME};
+    const CVideoInfoTag tag{*item.GetVideoInfoTag()};
+
+    // Get individual stack part as item
+    bool stackSet{false};
+    if (!options.state.empty())
+    {
+      CXBMCTinyXML2 xmlDoc;
+      if (xmlDoc.Parse(options.state))
+      {
+        tinyxml2::XMLHandle hRoot(xmlDoc.RootElement());
+        if (hRoot.ToElement() && StringUtils::EqualsNoCase(hRoot.ToElement()->Value(), "nextpart"))
+        {
+          int nextItem;
+          std::from_chars(hRoot.ToElement()->GetText(),
+                          hRoot.ToElement()->GetText() + std::strlen(hRoot.ToElement()->GetText()),
+                          nextItem);
+          item = stackHelper->SetStackPartCurrentFileItem(nextItem);
+          stackSet = true;
+          options.startpercent = options.starttime = 0;
+        }
+      }
+    }
+    if (!stackSet)
+    {
+      item = stackHelper->GetCurrentStackPartFileItem();
+
+      // Save resume state (needed for correct bluray playback)
+      if (Resume)
+        item.SetStartOffset(STARTOFFSET_RESUME);
+
+      // Adjust bookmark start
+      options.starttime = stackOffset;
+    }
+
+    // Keep VideoInfoTag
+    const std::string path = item.GetPath();
+    item.SetFromVideoInfoTag(tag);
+    item.SetPath(path);
```

Comment:
This reads like a hack. Saving the path just to re-apply it after something not really wanted happened, obviously?

Can't you just create an new item using `CFileItem::CFileItem(const CVideoInfoTag& movie)`?